api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -79,6 +79,7 @@ type NodePool struct {
 // NodePoolSpec is the desired behavior of a NodePool.
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.arch) || has(self.arch)", message="Arch is required once set"
 // +kubebuilder:validation:XValidation:rule="self.arch != 'arm64' || has(self.platform.aws)", message="Setting Arch to arm64 is only supported for AWS"
+// +kubebuilder:validation:XValidation:rule="has(self.replicas) ? !has(self.autoScaling) : has(self.autoScaling)", message="One of replicas or autoScaling should be set but not both"
```

Comment:
right, I haven't thought of that! but I think its better to have one as required, no?
Also, why is replicas a pointer if we default it to 0 anyway, there is not different between explicitly setting 0 and nil.

Reply:
It is only defaulted to 0 if `autoScaling` is not set.  It is a pointer so you can have `replicas: nil` and `autoScaling` set.

Reply:
are we expecting the user to have something like:
```yaml
spec:
  replicas: nil
  autoscaling :
    min: 1
    max: 2
```
rather than removing the replicas field completely from the spec? if we don't care to support this syntax, I don't think we need a pointer for replicas.

Reply:
We want it to be a pointer to make the distinction between "the user didn't set this" vs "the user did set it and set it to 0".  `replicas` is a pointer in all scalable resources that I know of.

Reply:
right, but those resources usually have a default > 0 when replicas is not set which is why the distinction is important but our default is 0 which is why I find it weird.